maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/synccontext/named/BasedirNameMapperTest.java

```diff
@@ -59,6 +60,42 @@ public void nullsAndEmptyInputs()
         assertThat( names, Matchers.empty() );
     }
 
+    @Test
+    public void defaultLocksDirName()
+    {
+        configProperties.put( "aether.syncContext.named.hashing.depth", "0" );
+        configProperties.put( "aether.syncContext.named.basedir.locksDirName", null );
+        DefaultArtifact artifact = new DefaultArtifact( "group:artifact:1.0" );
+        Collection<String> names = mapper.nameLocks( session, singletonList( artifact ), null );
+        assertThat( names, hasSize( 1 ) );
+        assertThat( names.iterator().next(),
+                equalTo( baseDir + PS + ".locks" + PS + "46e98183d232f1e16f863025080c7f2b9797fd10" ) );
+    }
+
+    @Test
+    public void relativeLocksDirName()
+    {
+        configProperties.put( "aether.syncContext.named.hashing.depth", "0" );
+        configProperties.put( "aether.syncContext.named.basedir.locksDirName", "my/locks" );
+        DefaultArtifact artifact = new DefaultArtifact( "group:artifact:1.0" );
+        Collection<String> names = mapper.nameLocks( session, singletonList( artifact ), null );
+        assertThat( names, hasSize( 1 ) );
+        assertThat( names.iterator().next(),
+                equalTo( baseDir + PS + "my" + PS + "locks" + PS + "46e98183d232f1e16f863025080c7f2b9797fd10" ) );
+    }
+
+    @Test
+    public void absoluteLocksDirName()
+    {
+        configProperties.put( "aether.syncContext.named.hashing.depth", "0" );
+        configProperties.put( "aether.syncContext.named.basedir.locksDirName", "/my/locks" );
+        DefaultArtifact artifact = new DefaultArtifact( "group:artifact:1.0" );
+        Collection<String> names = mapper.nameLocks( session, singletonList( artifact ), null );
+        assertThat( names, hasSize( 1 ) );
+        assertThat( names.iterator().next(), // ends with as we do not test drive letter on non-Win plaf
+                endsWith( PS + "my" + PS + "locks" + PS + "46e98183d232f1e16f863025080c7f2b9797fd10" ) );
+    }
```

Comment:
I also consider this test as invalid since `Path` will combine to a relative path here: https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap04.html#tag_04_11